### PR TITLE
build: Update target SDK to API 31

### PR DESCRIPTION
**Pull request by corycaywood**
# Description
Also needed to update Ad Mob SDK to 20.5.0 because previous versions do not support API 31. See https://developers.google.com/admob/android/rel-notes Also needed to update the Kotlin version in order to support API 31.

## Links
MINI-4600

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Review by Climbatize — APPROVED** [review]

**Review by khairul-alam-licon — APPROVED** [review]
Looks good 👍

File: testapp/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -20,7 +20,8 @@
 
         <activity
             android:name="com.rakuten.tech.mobile.testapp.ui.miniapplist.MiniAppListActivity"
-            android:launchMode="singleTop">
+            android:launchMode="singleTop"
+            android:exported="true">
```
Comment (munir-rakuten) [inline testapp/src/main/AndroidManifest.xml:24]: @corycaywood i am curious why MiniAppListActivity need `exported = true`, is it supposed to be accessed by outside of  Demo App?

**Review by munir-rakuten — APPROVED** [review]
Looks good to me 👍, I have just left a comment out of curiosity and also i think the jdk version of the container needs to update to pass the CI check.. thanks

**Comment by corycaywood** [comment]
Closing this PR. 
Replaced by https://github.com/rakutentech/android-miniapp/pull/413

**Pull request closed without merging**